app/src/main/java/ch/epfl/sweng/erpa/LoginActivity.java

```diff
@@ -0,0 +1,90 @@
+package ch.epfl.sweng.erpa;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+import ch.epfl.sweng.erpa.model.User;
+
+public class LoginActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+    }
+
+    public void login(View view) {
+        String usernameText = ((EditText) findViewById(R.id.username)).getText().toString();
+        String passwordText = ((EditText) findViewById(R.id.password)).getText().toString();
+
+        if (usernameText.isEmpty()) {
+            createPopup(getString(R.string.noNameMessage));
+            return;
+        } else if (passwordText.isEmpty()) {
+            createPopup(getString(R.string.noPassMessage));
+            return;
+        }
+
+        if (!checkLogin(usernameText, passwordText)) {
+            createPopup(getString(R.string.incorrectLogin));
+            return;
+        }
+
+        Intent intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
+    }
+
+    public void continueWithoutLogin(View view) {
+        Intent intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
+    }
+
+    private void createPopup(String text) {
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+
+        final TextView tv = new TextView(this);
+        tv.setText(text);
+        tv.setTextColor(Color.RED);
+        tv.setTextSize(16);
+
+        // set prompts.xml to alertdialog builder
+        alertDialogBuilder.setView(tv);
+
+        // set dialog message
+        alertDialogBuilder.setCancelable(false).setPositiveButton("OK", new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int id) {
+            }
+        });
+
+        // create alert dialog
+        AlertDialog alertDialog = alertDialogBuilder.create();
+        // show it
+        alertDialog.show();
+    }
+
+    private boolean checkLogin(String name, String pass) {
+        User toCheck = new User(name, pass);
+        ArrayList<User> userList = getUserList();
+        for(User u : userList) {
+            if (toCheck.equals(u))
+                return true;
+        }
+        return false;
+    }
+
+    private ArrayList<User> getUserList() {
+        ArrayList<User> userList = new ArrayList<>();
+        userList.add(new User("admin", "admin"));
```

Comment:
wtf? This shouldn't be here. If you need to “Preload” information, please use resources and a brokering service.

e.g. You could create something like “UserAuthenticator” that takes a username and password, converts the password to an access token (identity function would be okay for now) and then checks them (Hardcoded values like here would be a good start).